src/App.jsx

```diff
@@ -1,76 +1,135 @@
-    itemStatus: false,
-    id: generateUniqueID()
-  },
-  {
-    itemName: 'another item',
-    itemStatus: false,
-    id: generateUniqueID()
-  },
-  {
-    itemName: 'and another item',
-    itemStatus: true,
-    id: generateUniqueID()
-  }
-];
+// const INITIAL_DATA = [
+//   {
+//     itemName: "default item",
+//     itemStatus: false,
+//     id: generateUniqueID(),
+//   },
+//   {
+//     itemName: "another item",
+//     itemStatus: false,
+//     id: generateUniqueID(),
+//   },
+//   {
+//     itemName: "and another item",
+//     itemStatus: true,
+//     id: generateUniqueID(),
+//   },
+// ];
 
 const App = () => {
-  // TO ASK should this be in app or todolist
-  // here, todoitems update -> whole list rerender?
-  // in todolist, only updated items rerender?
-  const [todoItems, setTodoItems] = useState(INITIAL_DATA);
+  // const [todoItems, setTodoItems] = useState(INITIAL_DATA);
+  const TODO_ITEMS_PATH = "/users/test-user/todo-items";
+  const todoItemsRef = collection(db, TODO_ITEMS_PATH);
+  const [todoItems, loading] = useCollectionData(todoItemsRef);
   const [isAddClicked, setIsAddClicked] = useState(false);
-  const [newItemNameInput, setNewItemNameInput] = useState('');
+  const [newItemNameInput, setNewItemNameInput] = useState("");
   const [isCompletedShown, setIsCompletedShown] = useState(true);
+  const [priorityInput, setPriorityInput] = useState(0);
 
   const handleItemChange = (id, field, value) => {
-    const newTodoItems = todoItems.map((item) => {
-      return item.id === id ? { ...item, [field]: value } : item;
+    // const newTodoItems = todoItems.map((item) => {
+    //   return item.id === id ? { ...item, [field]: value } : item;
+    // });
+    // setTodoItems(newTodoItems);
+    const changedDocRef = doc(db, `${TODO_ITEMS_PATH}/${id}`);
+    updateDoc(changedDocRef, {
+      [field]: value,
     });
-    setTodoItems(newTodoItems);
   };
 
   const handleItemAdd = (name) => {
-    setTodoItems([...todoItems, { itemName: name, itemStatus: false, id: generateUniqueID() }]);
+    // setTodoItems([
+    //   ...todoItems,
+    //   { itemName: name, itemStatus: false, id: generateUniqueID() },
+    // ]);
+    const id = generateUniqueID();
+    const newDocRef = doc(db, `${TODO_ITEMS_PATH}/${id}`);
+    setDoc(newDocRef, {
+      itemName: name,
+      itemStatus: false,
+      id,
+    });
     setIsAddClicked(false);
-    setNewItemNameInput('');
+    setNewItemNameInput("");
   };
 
   const handleToggleShowCompleted = () => {
     setIsCompletedShown((prevState) => !prevState);
   };
 
-  const handleDeleteCompleted = () => {
-    const newData = todoItems.filter((item) => item.itemStatus === false);
-    setTodoItems(newData);
+  const handleDeleteCompleted = async () => {
+    // const newData = todoItems.filter((item) => item.itemStatus === false);
+    // setTodoItems(newData);
+
+    // create a query with the required conditions
+    const completedItemsQuery = query(
```

Comment:
No need for this.  You already have a list of all the items in todoItems. Just filter on the status, and then you have the needed IDs.